src/api/version.ts

```diff
@@ -163,7 +176,151 @@ export namespace VersionHelper {
     }
   }
 
-  function removeVPrefix(version: string): string {
-    return version.startsWith('v') ? version.substring(1) : version
+  /**
+   * Returns latest chectl version for the given channel.
+   */
+  export async function getLatestChectlVersion(channel: string): Promise<string | undefined> {
+    if (getProjectName() !== 'chectl') {
+      return
+    }
+
+    const axiosInstance = axios.create({
+      httpsAgent: new https.Agent({})
+    })
+
+    try {
+      const { data } = await axiosInstance.get(`https://che-incubator.github.io/chectl/channels/${channel}/linux-x64`)
+      return data.version
+    } catch {
+      return
+    }
+  }
+
+  /**
+   * Checks whether there is an update available for current chectl.
+   */
+  export async function isChectlUpdateAvailable(cacheDir: string, forceRecheck = false): Promise<boolean> {
+    // Do not use ctx inside this function as the function is used from hook where ctx is not yet defined.
+
+    if (getProjectName() !== 'chectl') {
+      // Do nothing for chectl flavors
+      return false
+    }
+
+    const currentVersion = getProjectVersion()
+    if (currentVersion === CHECTL_DEVELOPMENT_VERSION) {
+      // Skip it, chectl is built from source
+      return false
+    }
+
+    const channel = currentVersion.includes('next') ? 'next' : 'stable'
+    const newVersionInfoFilePath = path.join(cacheDir, `${channel}-${UPDATE_INFO_FILENAME}`)
+    let newVersionInfo: NewVersionInfoData = {
+      latestVersion: '0.0.0',
+      lastCheck: 0,
+    }
+    if (fs.existsSync(newVersionInfoFilePath)) {
+      newVersionInfo = JSON.parse(fs.readFileSync(newVersionInfoFilePath, 'utf8')) as NewVersionInfoData
+    }
+
+    const now = Date.now()
+    if (forceRecheck || now - newVersionInfo.lastCheck > A_DAY_IN_MS) {
```

Comment:
We do need to store the info for sure. It will prevent from making a lot of request (without it we have to do at least 1 request per each chectl command invocation)